src/com/mambu/apisdk/util/RequestExecutorImpl.java

```diff
@@ -139,21 +141,22 @@ public String executeRequest(String urlString, ParamsMap params, Method method,
 		} catch (IOException e) {
 			LOGGER.warning("IOException: message= " + e.getMessage());
 			throw new MambuApiException(e);
+		} finally {
+			httpClient.getConnectionManager().shutdown();
```

Comment:
I'd assume if we move the httpClient object to a field and reuse it everywhere, the lib takes care of connection persistence itself (http://hc.apache.org/httpclient-3.x/performance.html#Connection_persistence).